Rocky.agent.lib.nut

```diff
@@ -0,0 +1,1084 @@
+     *
+     * @param {object} context        - The Rocky.Context containing the request.
+     * @param {Array} defaultHandlers - The currently registered handlers.
+     *
+    */
+    function execute(context, defaultHandlers) {
+        // NOTE: Copying these handlers into the route might have some unintended side effect.
+        //       Consider changing this if issues come up.
+        foreach (handlerName, handler in defaultHandlers) {
+            // Copy over the non-middleware handlers
+            if (handlerName != "middlewares") {
+                if (!hasHandler(handlerName)) { _setHandler(handlerName, handler); }
+            } else {
+                // Copy the handlers over so we can iterate through in
+                // the correct order:
+                for (local i = handler.len() -1; i >= 0; i--) {
+                    // Only add handlers that we haven't already added
+                    if (_handlers.middlewares.find(handler[i]) == null) {
+                        _handlers.middlewares.insert(0, handler[i]);
+                    }
+                }
+            }
+        }
+
+        // Run all the handlers
+        _invokeNextHandler(context);
+    }
+
+    /**
+     * Register a route-level authorization handler.
+     *
+     * @param {function} callback - The handler that will process route-level requests.
+     *
+     * @returns {object} The target Rocky.route instance (this).
+    */
+    function authorize(callback) {
+        if (typeof callback != "function") throw ROCKY_ERROR.BAD_CALLBACK;
+        return _setHandler("authorize", callback);
+    }
+
+    /**
+     * Register a route-level 'request rejected' handler.
+     *
+     * @param {function} callback - The handler that will process route-level requests.
+     *
+     * @returns {object} The target Rocky.route instance (this).
+    */
+    function onUnauthorized(callback) {
+        if (typeof callback != "function") throw ROCKY_ERROR.BAD_CALLBACK;
+        return _setHandler("onUnauthorized", callback);
+    }
+
+    /**
+     * Register a route-level authorization handler.
+     *
+     * @param {function} callback - The handler that will process route-level requests.
+     *
+     * @returns {object} The target Rocky.route instance (this).
+    */
+    function onException(callback) {
+        if (typeof callback != "function") throw ROCKY_ERROR.BAD_CALLBACK;
+        return _setHandler("onException", callback);
+    }
+
+    /**
+     * Register a route-level request timeout handler.
+     *
+     * @param {function} callback - The handler that will process route-level requests.
+     * @param {integer}  timeout  - The timeout period in seconds.
+     *
+     * @returns {object} The target Rocky.route instance (this).
+    */
+    function onTimeout(callback, timeout = null) {
+        if (typeof callback != "function") throw ROCKY_ERROR.BAD_CALLBACK;
+        if (timeout == null) timeout = _timeout;
+        _timeout = timeout;
```

Comment:
I think these 2 lines can be simplified to: 
if ( timeout != null ) _timeout = timeout;